my-vault/internal/handlers/handlers.go

```diff
@@ -0,0 +1,181 @@
+
+	//var secret Secret
+	var secretRequestObject CreateSecretRequest
+
+	err := json.NewDecoder(r.Body).Decode(&secretRequestObject)
+	defer r.Body.Close()
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		log.Printf("ERROR: Unexpected behaviour: %v", err)
+		return
+	}
+	// build new secretObject
+	temp := NewCreateSecret(secretRequestObject)
+	//err = db.PutNewSecret(temp)
+	err = h.DB.PutNewSecret(temp)
+
+	if err != nil {
+
+		err := errors.New("unexpected internal error")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Printf("ERROR: Unexpted internal error: %v", err)
+		return
+	}
+
+	res := CreateSecretResponse{
+		Id:      temp.Id,
+		Name:    temp.Name,
+		Message: fmt.Sprintf("Successfully created secret with secretId: %s and name: %s", temp.Id, temp.Name),
+	}
+	
+	log.Printf("INFO: Secret saved----------------------------------------- %s", temp.Id)
+	w.Header().Set("Content-Type", "application/json")
+	encoder := json.NewEncoder(w)
+	err = encoder.Encode(res)
+	if err != nil {
+
+		err := errors.New("unexpected internal error")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Printf("ERROR: Unexpted internal error: %v", err)
+		return
+	}
+}
+
+func (h *SecretHandler) HandleGetSecretById(w http.ResponseWriter, r *http.Request) {
+	//curl  -X POST -H "Content-Type: application/json" http://localhost:9000/secrets/234
+	secretID := r.PathValue("secret_id")
+	secretEntry, err := h.DB.GetScretsById(secretID)
+	
+	if err.Error() == ErrNotFound.Error(){
+		http.Error(w, fmt.Sprintf("Secret with ID %s not found", secretID), http.StatusNotFound)
+		return
+	} else if err != nil{
+		http.Error(w, "Unexpected internal error", http.StatusInternalServerError)
+		log.Printf("ERROR: Unexpted internal error: %v", err)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	getSecretResponse := NewSecretResponseObject(secretEntry)
+	json.NewEncoder(w).Encode(getSecretResponse)
+	fmt.Printf("Retrieved secret----------------------------------------- %s", getSecretResponse)
+}
+
+func (h *SecretHandler) HandleDeleteSecretById(w http.ResponseWriter, r *http.Request) {
+	//curl  -X POST -H "Content-Type: application/json" http://localhost:9000/secrets/234
+	secretId := r.PathValue("secret_id")
+	secret, err := h.DB.DeleteSecretByID(secretId)
+	if err.Error()==ErrNotFound.Error(){
+		http.Error(w, fmt.Sprintf("Secret with ID %s not found", secret), http.StatusNotFound)
+		return
+	}
+	if err != nil {
+		log.Printf("Failed to delete secret: %v", err)
+		return
+	}
+	log.Println("Deleted secret with id -----------------------------------------", secret)
+	fmt.Fprintf(w, "Secret with ID %s deleted successfully", secret)
+}
+
+func (h *SecretHandler) HandleListSecrets(w http.ResponseWriter, r *http.Request) {
+
+	secrets, err := h.DB.ListAllSecrets()
+	if err.Error() == ErrNotFound.Error(){
+		http.Error(w,"No secrets found in the vault", http.StatusNotFound)
+		return
```

Comment:
I still don't think this should be an error. Listing when there are no secrets is totally valid.